File: android-staff/app/src/main/java/com/autoflow/staff/StaffPrefs.kt
package com.autoflow.staff

import android.content.Context
import java.util.UUID

object StaffPrefs {
    private const val PREFS = "autoflow_staff_native"
    private const val KEY_FCM_TOKEN = "fcm_token"
    private const val KEY_INVITE_TOKEN = "invite_token"
    private const val KEY_DEVICE_KEY = "device_key"
    private const val KEY_LAST_REGISTER_OK_AT = "last_register_ok_at"

    fun setFcmToken(context: Context, token: String) {
        context.prefs().edit().putString(KEY_FCM_TOKEN, token).apply()
    }

    fun getFcmToken(context: Context): String {
        return context.prefs().getString(KEY_FCM_TOKEN, "").orEmpty()
    }

    fun setInviteToken(context: Context, token: String) {
        context.prefs().edit().putString(KEY_INVITE_TOKEN, token).apply()
    }

    fun getInviteToken(context: Context): String {
        return context.prefs().getString(KEY_INVITE_TOKEN, "").orEmpty()
    }

    fun getOrCreateDeviceKey(context: Context): String {
        val prefs = context.prefs()
        val existing = prefs.getString(KEY_DEVICE_KEY, "").orEmpty()
        if (existing.isNotBlank()) return existing
        val created = "android_${UUID.randomUUID()}"
        prefs.edit().putString(KEY_DEVICE_KEY, created).apply()
        return created
    }

    fun setLastRegisterOkAt(context: Context, value: Long) {
        context.prefs().edit().putLong(KEY_LAST_REGISTER_OK_AT, value).apply()
    }

    private fun Context.prefs() = getSharedPreferences(PREFS, Context.MODE_PRIVATE)
}

File: ios-staff/AutoFlowStaff/Services/StaffPrefs.swift
import Foundation

/// Native prefs mirror android-staff/StaffPrefs.kt keys and semantics.
enum StaffPrefs {
    private static let suite = "autoflow_staff_native"

    private enum Key {
        static let fcmToken = "fcm_token"
        static let inviteToken = "invite_token"
        static let sessionToken = "session_token"
        static let deviceKey = "device_key"
        static let lastRegisterOkAt = "last_register_ok_at"
    }

    private static var defaults: UserDefaults { UserDefaults(suiteName: suite) ?? .standard }

    static func setFcmToken(_ token: String) {
        defaults.set(token, forKey: Key.fcmToken)
    }

    static func getFcmToken() -> String {
        defaults.string(forKey: Key.fcmToken) ?? ""
    }

    static func setInviteToken(_ token: String) {
        defaults.set(token, forKey: Key.inviteToken)
    }

    static func getInviteToken() -> String {
        defaults.string(forKey: Key.inviteToken) ?? ""
    }

    static func setSessionToken(_ token: String) {
        defaults.set(token, forKey: Key.sessionToken)
    }

    static func getSessionToken() -> String {
        defaults.string(forKey: Key.sessionToken) ?? ""
    }

    static func clearSessionToken() {
        defaults.removeObject(forKey: Key.sessionToken)
    }

    static func getOrCreateDeviceKey() -> String {
        if let existing = defaults.string(forKey: Key.deviceKey), !existing.isEmpty {
            return existing
        }
        let created = "ios_\(UUID().uuidString)"
        defaults.set(created, forKey: Key.deviceKey)
        return created
    }

    static func setLastRegisterOkAt(_ ms: Int64) {
        defaults.set(ms, forKey: Key.lastRegisterOkAt)
    }
}
